### ISSUE TITLE
Doesn't work on 13.0-RELEASE

### ISSUE THREAD

**Issue by andoriyu** [phyber/jail_exporter#135]
This is due rctl crate not working on 13.0-RELEASE yet.

It's already fixed in main branch here: https://github.com/fubarnetes/rctl/commit/2073c35696fbe0e761ac6fca8147c3f545fb3c91

Just making this issue to make you aware that soon dependency crate need to be updated.

**Comment by fabianfreyer** [comment]
> This is due rctl crate not working on 13.0-RELEASE yet.
> 
> 
> 
> It's already fixed in main branch here: https://github.com/fubarnetes/rctl/commit/2073c35696fbe0e761ac6fca8147c3f545fb3c91

Yup, planning on getting that released and `jail` as soon as possible!

**Comment by phyber** [comment]
@fabianfreyer any ETA on new versions of `rctl` and `jail`? If they're a while off, I'll just pull in some specific patches from them to get `jail_exporter` working again :)

**Comment by yaazkal** [comment]
Oh, I guess this is why then I'm getting `Error: RctlUnavailable("Support not present in kernel; see rctl(8) for details")` on `13.0-RELEASE-p3`

So maybe on 13.1 should work?

**Comment by andoriyu** [comment]
> 
> 
> Oh, I guess this is why then I'm getting `Error: RctlUnavailable("Support not present in kernel; see rctl(8) for details")` on `13.0-RELEASE-p3`
> 
> So maybe on 13.1 should work?

It wouldn't work on any version of 13 branch until `rctl`  and `jail` crates get a new release.

**Comment by phyber** [comment]
I've opened an issue in `rctl` to try and get a release of the crate. If that doesn't happen, I might implement the `rctl` bits I need directly in `jail_exporter`.

Sorry for the wait here.

**Comment by yaazkal** [comment]
> It wouldn't work on any version of 13 branch until rctl and jail crates get a new release.
Oh ok, thanks for the clarification @andoriyu 

> I've opened an issue in rctl to try and get a release of the crate. If that doesn't happen, I might implement the rctl bits I need directly in jail_exporter.
Thank you so much @phyber

**Comment by phyber** [comment]
I've pushed some new code to the `master` branch so that `jail_exporter` can check `rctl` availability itself. After a few days of running this locally, I'll push a new version to crates.io and update the FreeBSD port.

**Comment by fabianfreyer** [comment]
Sorry for the long wait, I've released a new version of `rctl` and `jail`.

**Comment by phyber** [comment]
@fabianfreyer thanks a lot!

**Comment by phyber** [comment]
A new release has finally been tagged and pushed to crates.io. I have an update to the port locally, and I should be able to submit it a little later on.

**Comment by yaazkal** [comment]
> A new release has finally been tagged and pushed to crates.io. I have an update to the port locally, and I should be able to submit it a little later on.

thanks so much !

I hope it lands on binary ports soon to try again. Have a great week.
